packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -310,6 +313,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    public async tombstone() {
+        this.verifyNotClosed();
+        if (this.tombstoned) {
+            return;
+        }
+
+        assert(!(await this.isRoot()), "Tombstone should not be called on root datastores!");
+        assert(!this.isLocalDataStore, "Local data should have been wiped before tombstone is called!");
+
+        // Note loaded and channel !== undefined are the same thing as you are only loaded if the channel is defined
+        assert(!this.loaded || this.channel === undefined, "Cannot tombstone a loaded datastore.");
```

Comment:
Why can't a loaded data store be tombstoned?

Reply:
That would mean it's loaded into memory correct?

Reply:
The real question is that what should the behavior be when a loaded datastore is tombstoned? What happens to a loaded datastore when it's tombstoned? Shouldn't GC be ensuring that it's only deleting datastores that aren't used?

Reply:
A data store can be loaded for any number of reasons - it is independent of GC or the fact that it is unreferenced. We cannot assume that a data store that is sweep ready must not be loaded.
For example, summarize() can be called with fullTree = true which will result in loading all data stores. We should be able to tombstone such data stores as well. Another example is someone requesting an unreferenced data store **_before_** it is sweep ready but never using it and such data stores can be eligible for tombstones.